### ISSUE TITLE
[curl] Vcpkg install libcurl error

### ISSUE THREAD

**Issue by ChocolateJin** [microsoft/vcpkg#9423]
I tried to install libcurl but i failed. Plz help me to find out a solution.
Here is the screenshot
![image](https://user-images.githubusercontent.com/56467837/71363198-5a356f80-25d3-11ea-955a-909920022eb6.png)

**Comment by ChocolateJin** [comment]
addition
![image](https://user-images.githubusercontent.com/56467837/71363774-2c512a80-25d5-11ea-8955-8fd4be3978ab.png)

**Comment by PhoebeHui** [comment]
Could you provide us config-x86-windows-out.log?

**Comment by ChocolateJin** [comment]
> Could you provide us config-x86-windows-out.log?
Sorry.I forgot it.Here's the file content.


[1/2] cmd /c "cd ../../x86-windows-dbg && "C:/Program Files/CMake/bin/cmake.exe" "D:/vcpkg/buildtrees/curl/src/url-7_66_0-ac7576f585" "-DBUILD_TESTING=OFF" "-DBUILD_CURL_EXE=OFF" "-DENABLE_MANUAL=OFF" "-DCURL_STATICLIB=0" "-DCMAKE_USE_OPENSSL=OFF" "-DCMAKE_USE_WINSSL=ON" "-DCMAKE_USE_MBEDTLS=OFF" "-DCMAKE_USE_SECTRANSP=OFF" "-DCMAKE_USE_LIBSSH2=OFF" "-DHTTP_ONLY=ON" "-DENABLE_ARES=OFF" "-DCURL_WINDOWS_SSPI=OFF" "-DCURL_BROTLI=OFF" "-DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v141" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-DBUILD_CURL_EXE=OFF" "-DENABLE_DEBUG=ON" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/curl_x86-windows/debug""
FAILED: ../../x86-windows-dbg/CMakeCache.txt 
cmd /c "cd ../../x86-windows-dbg && "C:/Program Files/CMake/bin/cmake.exe" "D:/vcpkg/buildtrees/curl/src/url-7_66_0-ac7576f585" "-DBUILD_TESTING=OFF" "-DBUILD_CURL_EXE=OFF" "-DENABLE_MANUAL=OFF" "-DCURL_STATICLIB=0" "-DCMAKE_USE_OPENSSL=OFF" "-DCMAKE_USE_WINSSL=ON" "-DCMAKE_USE_MBEDTLS=OFF" "-DCMAKE_USE_SECTRANSP=OFF" "-DCMAKE_USE_LIBSSH2=OFF" "-DHTTP_ONLY=ON" "-DENABLE_ARES=OFF" "-DCURL_WINDOWS_SSPI=OFF" "-DCURL_BROTLI=OFF" "-DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v141" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-DBUILD_CURL_EXE=OFF" "-DENABLE_DEBUG=ON" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Debug" "-DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/curl_x86-windows/debug""
-- The C compiler identification is unknown
-- Check for working C compiler: C:/Windows/gcc.exe
-- Check for working C compiler: C:/Windows/gcc.exe -- broken
CMake Error at C:/Program Files/CMake/share/cmake-3.16/Modules/CMakeTestCCompiler.cmake:60 (message):
  The C compiler

    "C:/Windows/gcc.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: D:/vcpkg/buildtrees/curl/x86-windows-dbg/CMakeFiles/CMakeTmp
    
    Run Build Command(s):C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe cmTC_241d1 && [1/2] Building C object CMakeFiles\cmTC_241d1.dir\testCCompiler.c.obj
    FAILED: CMakeFiles/cmTC_241d1.dir/testCCompiler.c.obj 
    C:\Windows\gcc.exe   /DWIN32 /D_WINDOWS /W3 /utf-8 /MP -o CMakeFiles\cmTC_241d1.dir\testCCompiler.c.obj   -c testCCompiler.c
    testCCompiler.c:2:3: #error "The CMAKE_C_COMPILER is set to a C++ compiler"
    g++.exe: /DWIN32: No such file or directory
    g++.exe: /D_WINDOWS: No such file or directory
    g++.exe: /W3: No such file or directory
    g++.exe: /utf-8: No such file or directory
    g++.exe: /MP: No such file or directory
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:48 (project)


-- Configuring incomplete, errors occurred!
See also "D:/vcpkg/buildtrees/curl/x86-windows-dbg/CMakeFiles/CMakeOutput.log".
See also "D:/vcpkg/buildtrees/curl/x86-windows-dbg/CMakeFiles/CMakeError.log".
[2/2] cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "D:/vcpkg/buildtrees/curl/src/url-7_66_0-ac7576f585" "-DBUILD_TESTING=OFF" "-DBUILD_CURL_EXE=OFF" "-DENABLE_MANUAL=OFF" "-DCURL_STATICLIB=0" "-DCMAKE_USE_OPENSSL=OFF" "-DCMAKE_USE_WINSSL=ON" "-DCMAKE_USE_MBEDTLS=OFF" "-DCMAKE_USE_SECTRANSP=OFF" "-DCMAKE_USE_LIBSSH2=OFF" "-DHTTP_ONLY=ON" "-DENABLE_ARES=OFF" "-DCURL_WINDOWS_SSPI=OFF" "-DCURL_BROTLI=OFF" "-DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v141" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-DBUILD_CURL_EXE=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/curl_x86-windows""
FAILED: ../CMakeCache.txt 
cmd /c "cd .. && "C:/Program Files/CMake/bin/cmake.exe" "D:/vcpkg/buildtrees/curl/src/url-7_66_0-ac7576f585" "-DBUILD_TESTING=OFF" "-DBUILD_CURL_EXE=OFF" "-DENABLE_MANUAL=OFF" "-DCURL_STATICLIB=0" "-DCMAKE_USE_OPENSSL=OFF" "-DCMAKE_USE_WINSSL=ON" "-DCMAKE_USE_MBEDTLS=OFF" "-DCMAKE_USE_SECTRANSP=OFF" "-DCMAKE_USE_LIBSSH2=OFF" "-DHTTP_ONLY=ON" "-DENABLE_ARES=OFF" "-DCURL_WINDOWS_SSPI=OFF" "-DCURL_BROTLI=OFF" "-DCMAKE_DISABLE_FIND_PACKAGE_Perl=ON" "-DCMAKE_MAKE_PROGRAM=C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe" "-DBUILD_SHARED_LIBS=ON" "-DVCPKG_CHAINLOAD_TOOLCHAIN_FILE=D:/vcpkg/scripts/toolchains/windows.cmake" "-DVCPKG_TARGET_TRIPLET=x86-windows" "-DVCPKG_SET_CHARSET_FLAG=ON" "-DVCPKG_PLATFORM_TOOLSET=v141" "-DCMAKE_EXPORT_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_PACKAGE_REGISTRY=ON" "-DCMAKE_FIND_PACKAGE_NO_SYSTEM_PACKAGE_REGISTRY=ON" "-DCMAKE_INSTALL_SYSTEM_RUNTIME_LIBS_SKIP=TRUE" "-DCMAKE_VERBOSE_MAKEFILE=ON" "-DVCPKG_APPLOCAL_DEPS=OFF" "-DCMAKE_TOOLCHAIN_FILE=D:/vcpkg/scripts/buildsystems/vcpkg.cmake" "-DCMAKE_ERROR_ON_ABSOLUTE_INSTALL_DESTINATION=ON" "-DVCPKG_CXX_FLAGS=" "-DVCPKG_CXX_FLAGS_RELEASE=" "-DVCPKG_CXX_FLAGS_DEBUG=" "-DVCPKG_C_FLAGS=" "-DVCPKG_C_FLAGS_RELEASE=" "-DVCPKG_C_FLAGS_DEBUG=" "-DVCPKG_CRT_LINKAGE=dynamic" "-DVCPKG_LINKER_FLAGS=" "-DVCPKG_TARGET_ARCHITECTURE=x86" "-DCMAKE_INSTALL_LIBDIR:STRING=lib" "-DCMAKE_INSTALL_BINDIR:STRING=bin" "-DBUILD_CURL_EXE=OFF" "-G" "Ninja" "-DCMAKE_BUILD_TYPE=Release" "-DCMAKE_INSTALL_PREFIX=D:/vcpkg/packages/curl_x86-windows""
-- The C compiler identification is unknown
-- Check for working C compiler: C:/Windows/gcc.exe
-- Check for working C compiler: C:/Windows/gcc.exe -- broken
CMake Error at C:/Program Files/CMake/share/cmake-3.16/Modules/CMakeTestCCompiler.cmake:60 (message):
  The C compiler

    "C:/Windows/gcc.exe"

  is not able to compile a simple test program.

  It fails with the following output:

    Change Dir: D:/vcpkg/buildtrees/curl/x86-windows-rel/CMakeFiles/CMakeTmp
    
    Run Build Command(s):C:/Program Files (x86)/Microsoft Visual Studio/2017/Community/Common7/IDE/CommonExtensions/Microsoft/CMake/Ninja/ninja.exe cmTC_3e997 && [1/2] Building C object CMakeFiles\cmTC_3e997.dir\testCCompiler.c.obj
    FAILED: CMakeFiles/cmTC_3e997.dir/testCCompiler.c.obj 
    C:\Windows\gcc.exe   /DWIN32 /D_WINDOWS /W3 /utf-8 /MP -o CMakeFiles\cmTC_3e997.dir\testCCompiler.c.obj   -c testCCompiler.c
    testCCompiler.c:2:3: #error "The CMAKE_C_COMPILER is set to a C++ compiler"
    g++.exe: /DWIN32: No such file or directory
    g++.exe: /D_WINDOWS: No such file or directory
    g++.exe: /W3: No such file or directory
    g++.exe: /utf-8: No such file or directory
    g++.exe: /MP: No such file or directory
    ninja: build stopped: subcommand failed.
    
    

  

  CMake will not be able to correctly generate this project.
Call Stack (most recent call first):
  CMakeLists.txt:48 (project)


-- Configuring incomplete, errors occurred!
See also "D:/vcpkg/buildtrees/curl/x86-windows-rel/CMakeFiles/CMakeOutput.log".
See also "D:/vcpkg/buildtrees/curl/x86-windows-rel/CMakeFiles/CMakeError.log".
ninja: build stopped: subcommand failed.

**Comment by NancyLi1013** [comment]
Hi @ChocolateJin thanks for posting this issue.
I noticed that you installed this port on MinGW64.
Currently, vcpkg doesn't support MinGW.
You can use powershell to install and build port on Windows platform.

**Comment by ChocolateJin** [comment]
> Hi @ChocolateJin thanks for posting this issue.
> I noticed that you installed this port on MinGW64.
> Currently, vcpkg doesn't support MinGW.
> You can use powershell to install and build port on Windows platform.

Thx for your help. Do you mean that I should run command"./vcpkg install curl" on Powershell?
I tried but still failed. Just like this.
![image](https://user-images.githubusercontent.com/56467837/71399310-28b5b600-265e-11ea-8947-ad74f1886660.png)
I have run "./vcpkg update" before,Yet I failed again. Are there other solutions for this problem? Or should I uninstall MingGW64?

**Comment by NancyLi1013** [comment]
Could you please clone vcpkg on a clean environment and build this port again?
I am not sure if this issue is related to MingGW64.
I can install curl on my local successfully with PowerShell.
If still failed, please attach the error log.

Thanks.

**Comment by ChocolateJin** [comment]
> 您能否在干净的环境中克隆vcpkg并再次构建此端口？
> 我不确定此问题是否与MingGW64有关。
> 我可以使用PowerShell在本地成功安装curl。
> 如果仍然失败，请附加错误日志。
> 
> 谢谢。

OK,I'

> Could you please clone vcpkg on a clean environment and build this port again?
> I am not sure if this issue is related to MingGW64.
> I can install curl on my local successfully with PowerShell.
> If still failed, please attach the error log.
> 
> Thanks.

OK.I'm try it later.

**Comment by NancyLi1013** [comment]
@ChocolateJin  Any updates about this issue?
Please let me know if you still have problems.

**Comment by ChocolateJin** [comment]
> @ChocolateJin Any updates about this issue?
> Please let me know if you still have problems.

Sorry. My final exam is coming so I nearly have no time to test this problem. I'll try installing Vcpkg on a virtual machine tonight and I'll reply you later. Thank you for your caring.

**Comment by ChocolateJin** [comment]
> @ChocolateJin Any updates about this issue?
> Please let me know if you still have problems.

Oh,right. I've tried to reinstall cmake,vcpkg,mingw(together with gcc/g++). Yet the problem is still there. Here is the result(On Powershell 7-preview(x64))
![image](https://user-images.githubusercontent.com/56467837/71575893-931c9980-2b29-11ea-9cae-ac0d7b2defbd.png)
And error file report.
[config-x86-windows-out.log](https://github.com/microsoft/vcpkg/files/4009387/config-x86-windows-out.log)

I tried to complie a single .c file by using gcc and it failed. Don't know whether it is the reason.
![image](https://user-images.githubusercontent.com/56467837/71576192-b431ba00-2b2a-11ea-9961-de9130f3bdc0.png)

**Comment by PhoebeHui** [comment]
@ChocolateJin, mingw is not supported and tested by default so far, if you are interested in building libraries with mingw, see https://github.com/microsoft/vcpkg/issues/4418#issuecomment-428168157 for reference.

**Comment by ChocolateJin** [comment]
> @ChocolateJin, mingw is not supported and tested by default so far, if you are interested in building libraries with mingw, see [#4418 (comment)](https://github.com/microsoft/vcpkg/issues/4418#issuecomment-428168157) for reference.

emmmm.. I'm a newcomer on programming. If I can't use mingw to complete this task. Could you tell me what else substitutes to replace mingw on Windows? Thx.

**Comment by PhoebeHui** [comment]
We recommend trying out WSL to build binaries for Linux on a Windows machine.

**Comment by ChocolateJin** [comment]
I finally finished this problem. First I uninstall all mingw parts and add Visual Studio related configuration to system variable. Then I found the problem is probably is because this:
Command failed: ninja -v(It means Cmake can't complie through ninja)
Thus I found another quota   https://github.com/microsoft/vcpkg/issues/2895
After delecting related sentences. Cmake doesn't use ninja to compile. And the problem has been solved. Thx for your all help. Sincere Thanks. But my friend don't need to add variable or delect the sentences. The specific reason is still not clear.
